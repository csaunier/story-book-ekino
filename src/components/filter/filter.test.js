import React from 'react'

import { shallow } from 'enzyme'

import Filter from './filter'

jest.mock('shortid', () => ({ generate: () => '' }))

const props1 = {
  selectedFilter: '',
  selectFilter: jest.fn(),
}

const props2 = {
  selectedFilter: 'senior',
  selectFilter: jest.fn(),
}

describe('component Title', () => {
  it('should render by default with all member', () => {
    const component = shallow(<Filter {...props1} />)

    expect(component).toMatchSnapshot()
  })

  it('should render with a selected filter', () => {
    const component = shallow(<Filter {...props2} />)

    expect(component).toMatchSnapshot()
  })
})
