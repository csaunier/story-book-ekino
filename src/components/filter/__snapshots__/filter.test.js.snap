// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Title should render by default with all member 1`] = `
<div
  className="wrapper"
>
  <select
    className="select select--"
    onChange={[Function]}
    value=""
  >
    <option
      key="option--"
      value=""
    >
      L'union fait la force
    </option>
    <option
      key="option--"
      value="manitou"
    >
      Le manitou
    </option>
    <option
      key="option--"
      value="senior"
    >
      Les mecs qui pèsent
    </option>
    <option
      key="option--"
      value="archi"
    >
      Les mecs qui pèsent encore plus
    </option>
    <option
      key="option--"
      value="trainee"
    >
      Les pauvres
    </option>
    <option
      key="option--"
      value="true"
    >
      Les mauvais élèves
    </option>
    <option
      key="option--"
      value="london"
    >
      Pas besoin de cours d'anglais
    </option>
    <option
      key="option--"
      value="bordeaux"
    >
      Les adeptes du vin
    </option>
    <option
      key="option--"
      value="confirmed"
    >
      Enfin dans la cour des grands !
    </option>
    <option
      key="option--"
      value="junior"
    >
      Découverte du bac à sable
    </option>
    <option
      key="option--"
      value="female"
    >
      Le pouvoir aux femmes
    </option>
    <option
      key="option--"
      value="male"
    >
      La testostérone
    </option>
    <option
      key="option--"
      value="6"
    >
      L'empereur
    </option>
  </select>
</div>
`;

exports[`component Title should render with a selected filter 1`] = `
<div
  className="wrapper"
>
  <select
    className="select select--senior"
    onChange={[Function]}
    value="senior"
  >
    <option
      key="option--"
      value=""
    >
      L'union fait la force
    </option>
    <option
      key="option--"
      value="manitou"
    >
      Le manitou
    </option>
    <option
      key="option--"
      value="senior"
    >
      Les mecs qui pèsent
    </option>
    <option
      key="option--"
      value="archi"
    >
      Les mecs qui pèsent encore plus
    </option>
    <option
      key="option--"
      value="trainee"
    >
      Les pauvres
    </option>
    <option
      key="option--"
      value="true"
    >
      Les mauvais élèves
    </option>
    <option
      key="option--"
      value="london"
    >
      Pas besoin de cours d'anglais
    </option>
    <option
      key="option--"
      value="bordeaux"
    >
      Les adeptes du vin
    </option>
    <option
      key="option--"
      value="confirmed"
    >
      Enfin dans la cour des grands !
    </option>
    <option
      key="option--"
      value="junior"
    >
      Découverte du bac à sable
    </option>
    <option
      key="option--"
      value="female"
    >
      Le pouvoir aux femmes
    </option>
    <option
      key="option--"
      value="male"
    >
      La testostérone
    </option>
    <option
      key="option--"
      value="6"
    >
      L'empereur
    </option>
  </select>
</div>
`;
